### ISSUE TITLE
DJ008 incorrectly reporting missing `__str__` method for abstract class with type annotations

### ISSUE THREAD

**Issue by CarrotManMatt** [astral-sh/ruff#19154]
### Summary

When a Django model class is marked as abstract the lack of a `__str__` method is not flagged. This is correct behaviour.

```python
from django.db import models

class Foo(models.Model)
    class Meta:
        abstract = True
```

When a type annotation is added to the meta class, the identification of the model being abstract is lost, so the DJ008 diagnostic *incorrectly* occurs.

```python
from typing import ClassVar
from django.db import models
from django_stubs_ext.db.models import TypedModelMeta


class Foo(models.Model):
    class Meta(TypedModelMeta):
        abstract: ClassVar[bool] = True
```

### Version

0.12.2

**Comment by ntBre** [comment]
Thanks for the report! I think we should also check for a `Stmt::AnnAssign` here:

https://github.com/astral-sh/ruff/blob/2f8bc9a30bbcb3839fa0a5031e0bbae329210e53/crates/ruff_linter/src/rules/flake8_django/rules/model_without_dunder_str.rs#L99-L101
